src/hypothesis/strategies.py

```diff
@@ -686,12 +686,12 @@ def characters(whitelist_categories=None, blacklist_categories=None,
     if (
         whitelist_characters is not None and
         blacklist_characters is not None and
-        set(whitelist_characters).intersection(set(whitelist_characters))
+        set(blacklist_characters).intersection(set(whitelist_characters))
     ):
         raise InvalidArgument(
             'Cannot have characters in both whitelist_characters=%r, '
             'and blacklist_characters=%r' % (
```

Comment:
Not really part of this change, but while I'm looking at this... it would be nice to say why (e.g. something more like `"Characters %r are present both in whitelist_characters=%r and blacklist_characters=%r"`). Feel free to ignore this for now though.